File: Hangman.py
import tkinter as tk
from tkinter import messagebox

class HangmanGame:
    def __init__(self, master):
        self.master = master
        self.master.title("Hangman Game")
        
        self.word = input("Enter a word: ").lower()
        self.attempts = 15
        self.wrong = 0
        self.check = ["_"] * len(self.word)

        self.word_label = tk.Label(master, text=" ".join(self.check), font=("Helvetica", 18))
        self.word_label.pack(pady=10)

        self.attempts_label = tk.Label(master, text=f"Remaining Attempts: {self.attempts - self.wrong}", font=("Helvetica", 14))
        self.attempts_label.pack(pady=5)

        self.guess_entry = tk.Entry(master, font=("Helvetica", 14))
        self.guess_entry.pack(pady=5)

        self.submit_button = tk.Button(master, text="Submit", command=self.process_guess, font=("Helvetica", 14))
        self.submit_button.pack(pady=5)

    def process_guess(self):
        guess = self.guess_entry.get().lower()
        self.guess_entry.delete(0, tk.END)

        if len(guess) != 1:
            messagebox.showwarning("Invalid Input", "Please enter a single letter.")
            return

        if guess in self.word:
            for i, letter in enumerate(self.word):
                if letter == guess:
                    self.check[i] = guess
        else:
            self.wrong += 1

        self.word_label.config(text=" ".join(self.check))
        self.attempts_label.config(text=f"Remaining Attempts: {self.attempts - self.wrong}")

        if "".join(self.check) == self.word:
            messagebox.showinfo("Congratulations", "You Won the Game!")
            self.master.quit()
        elif self.wrong >= self.attempts:
            messagebox.showerror("Game Over", f"You Lost! The word was: {self.word}")
            self.master.quit()

if __name__ == "__main__":
    root = tk.Tk()
    game = HangmanGame(root)
    root.mainloop()

File: Hangman.java
// package Sara;

import javax.swing.*;
import java.awt.*;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import java.util.Arrays;

public class Hangman {

    private static final int MAX_ATTEMPTS = 13;
    private String word = "programming";
    private char[] guessedWord;
    private int wrongAttempts = 0;

    private JFrame frame;
    private JLabel wordLabel;
    private JLabel attemptsLabel;
    private JTextField inputField;
    private JLabel hangmanImage;

    public Hangman() {
        guessedWord = new char[word.length()];
        Arrays.fill(guessedWord, '_');
        
        frame = new JFrame("Hangman Game");
        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        frame.setSize(400, 400);
        frame.setLayout(new BorderLayout());

        wordLabel = new JLabel(new String(guessedWord), SwingConstants.CENTER);
        wordLabel.setFont(new Font("SansSerif", Font.BOLD, 24));

        attemptsLabel = new JLabel("Remaining Attempts: " + (MAX_ATTEMPTS - wrongAttempts), SwingConstants.CENTER);
        attemptsLabel.setFont(new Font("SansSerif", Font.PLAIN, 18));

        inputField = new JTextField();
        inputField.setHorizontalAlignment(JTextField.CENTER);
        inputField.setFont(new Font("SansSerif", Font.PLAIN, 18));

        hangmanImage = new JLabel();
        hangmanImage.setHorizontalAlignment(SwingConstants.CENTER);
        updateHangmanImage();

        inputField.addActionListener(new ActionListener() {
            @Override
            public void actionPerformed(ActionEvent e) {
                String input = inputField.getText().toLowerCase();
                if (input.length() == 1) {
                    processGuess(input.charAt(0));
                }
                inputField.setText("");
            }
        });

        frame.add(wordLabel, BorderLayout.NORTH);
        frame.add(attemptsLabel, BorderLayout.CENTER);
        frame.add(inputField, BorderLayout.SOUTH);
        frame.add(hangmanImage, BorderLayout.EAST);

        frame.setVisible(true);
    }

    private void processGuess(char guess) {
        boolean correctGuess = false;
        for (int i = 0; i < word.length(); i++) {
            if (word.charAt(i) == guess && guessedWord[i] == '_') {
                guessedWord[i] = guess;
                correctGuess = true;
            }
        }

        if (!correctGuess) {
            wrongAttempts++;
        }

        updateHangmanImage();
        wordLabel.setText(new String(guessedWord));
        attemptsLabel.setText("Remaining Attempts: " + (MAX_ATTEMPTS - wrongAttempts));

        if (String.valueOf(guessedWord).equals(word)) {
            JOptionPane.showMessageDialog(frame, "Congratulations! You've won the game!", "Game Over", JOptionPane.INFORMATION_MESSAGE);
            frame.dispose();
        } else if (wrongAttempts >= MAX_ATTEMPTS) {
            JOptionPane.showMessageDialog(frame, "Game Over! The word was: " + word, "Game Over", JOptionPane.ERROR_MESSAGE);
            frame.dispose();
        }
    }

    private void updateHangmanImage() {
        String imagePath = "C:\\Users\\pulik\\OneDrive\\Pictures\\profile" + wrongAttempts + ".jpg";
        ImageIcon icon = new ImageIcon(imagePath);
        hangmanImage.setIcon(icon);
    }

    public static void main(String[] args) {
        SwingUtilities.invokeLater(() -> new Hangman());
    }
}
